subsys/bluetooth/host/scan.c

```diff
@@ -24,14 +25,25 @@
 #include "id.h"
 
 #include "common/bt_str.h"
+#include "scan.h"
+#include <stdint.h>
+#include <string.h>
 
 #define LOG_LEVEL CONFIG_BT_HCI_CORE_LOG_LEVEL
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(bt_scan);
 
+struct scanner_state {
+	ATOMIC_DEFINE(scan_flags, SCAN_ENABLED_REASON_NUM_FLAGS);
+	struct bt_le_scan_param explicit_scan_param;
+	struct bt_le_scan_param used_scan_param;
+};
+
 static bt_le_scan_cb_t *scan_dev_found_cb;
 static sys_slist_t scan_cbs = SYS_SLIST_STATIC_INIT(&scan_cbs);
 
+static struct scanner_state m_scanner_state;
```

Comment:
What's with the `m_` prefix? 

Looks like something from Android that we don't use in Zephyr :)